### ISSUE TITLE
Generated a11y palettes failing when entering in #fff or #000 for all values

### ISSUE THREAD

**Issue by alex-page** [govau/design-system-components#607]
# Bug Report

- [x] I’ve read and understood the [Contributing guidelines](https://github.com/govau/design-system-components/blob/master/CONTRIBUTING.md) and have done my best effort to follow them.
- [x] I’ve read and agree to the [Code of Conduct](https://github.com/govau/design-system-components/blob/master/CODE_OF_CONDUCT.md).
- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

In our example we have:
```
$AU-color-foreground-action: #000;
$AU-color-foreground-text: #000;
$AU-color-background: #000;
```

We get the lowest contrast foreground color on the background entered by the user, in this example it would be `#000` as that is the item with the lowest contrast.

We then set the background color to be the closest passing accessible colour with the lowest contrast foreground `#000` on background `#000`. This makes the background `#767676`.

We then darken the background by 4%, 8% and 12%. The new darkened background fails as the text colour is dark and we started with the minimum passing colour contrast.

## What I expected to happen

We need to check if the new passing background colour when darkened by 12%  passes or fails on the lowest contrast foreground. The darkest variation of the background color is `#c2c2c2` which fails with a contrast of 2.87909214 contrast.

When this contrast fails, we need to increase the lightness of the original background color so that when we darken it the darkened version passes on the lowest contrast foreground color ( #000 ).

We can get the minimum lightness to generate a new hexadecimal colour by dividing 255 by 100 percents of lightness `255/100`.

```
if( contrast( darken( lighten(  background, 12 + 0.3692321% ), 12% ), lowest foreground ) > 4.5 ){
  return background
}
else {
  recursive lighten and if statement;
}
```

## Reproducing

- Module name: core
- Module version: 3.1.2
